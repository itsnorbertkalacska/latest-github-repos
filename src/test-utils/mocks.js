export const rawRepo = {
  id: 1,
  name: "My Repo",
  description: "This is the description",
  stargazersCount: 111,
  htmlUrl: "github.com/my-repo",
};

const rawRepoB = {
  id: 2,
  name: "Second Repo",
  description: "Lorem Ipsum",
  stargazersCount: 2450,
  htmlUrl: "github.com/this-is-second",
};

export const repo = {
  ...rawRepo,
  isFavourited: false,
};

const repoB = {
  ...rawRepoB,
  isFavourited: true,
};

export const rawRepos = [rawRepo, rawRepoB];

export const repos = [repo, repoB];
