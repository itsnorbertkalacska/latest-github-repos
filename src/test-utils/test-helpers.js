import { ThemeProvider } from "styled-components";
import { render } from "@testing-library/react";

import { theme } from "../theme";

export const renderWithProviders = (ui) => {
  function Wrapper({ children }) {
    return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
  }
  return render(ui, { wrapper: Wrapper });
};
