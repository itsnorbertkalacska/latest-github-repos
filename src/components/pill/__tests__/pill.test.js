import { renderWithProviders } from "../../../test-utils";
import { Pill } from "../index";

describe("Pill", () => {
  it("renders component properly", () => {
    const { container } = renderWithProviders(<Pill label="Fav" />);
    // eslint-disable-next-line testing-library/no-node-access
    expect(container.firstChild).toMatchSnapshot();
  });
});
