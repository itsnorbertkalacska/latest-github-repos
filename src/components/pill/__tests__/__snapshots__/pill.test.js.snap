// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pill renders component properly 1`] = `
<div
  class="sc-bczRLJ biJXxJ"
>
  <p>
    Fav
  </p>
</div>
`;
